frontend/src/AppRoot.tsx

```diff
@@ -52,6 +53,9 @@ const AppRoot = () => {
           <PrivateRoute exact isAuthenticated={isAuthenticated} path="/groups">
             <GroupList />
           </PrivateRoute>
+          <PrivateRoute isAuthenticated={isAuthenticated} path="*">
+            <NotFoundPage />
+          </PrivateRoute>
```

Comment:
This is a 404 page for users who are logged in. People who are logged out will be redirected in `PrivateRoute.tsx`.